data_validation/__main__.py

```diff
@@ -531,49 +531,49 @@ def run_validation(config_manager, dry_run=False, verbose=False):
         dry_run (bool): Print source and target SQL to stdout in lieu of validation.
         verbose (bool): Validation setting to log queries run.
     """
-    validator = DataValidation(
+    with DataValidation(
         config_manager.config,
         validation_builder=None,
         result_handler=None,
         verbose=verbose,
-    )
-
-    if dry_run:
-        sql_alchemy_clients = [
-            "mysql",
-            "oracle",
-            "postgres",
-            "db2",
-            "mssql",
-            "redshift",
-            "snowflake",
-        ]
-
-        source_query = validator.validation_builder.get_source_query().compile()
-        if config_manager.source_client.name in sql_alchemy_clients:
-            source_query = source_query.compile(
-                config_manager.source_client.con.engine,
-                compile_kwargs={"literal_binds": True},
-            )
+    ) as validator:
```

Comment:
I'm curious: was this change just for "styling" purposes or it was needed for the fix (that new __exit__ method) actually work?

Reply:
It ensures the new exit method is called. An alternative would have been to create a public exit or close method and explicitly call that. I think this is nicer.

Reply:
Indeed, I agree! Thanks for the explanation and great job 🚀